### PR TITLE
Fix position of the digitized SET strip hits in the SET

### PR DESCRIPTION
**Pull request by dudarboh**
BEGINRELEASENOTES

- Center of the SET strips is now properly in the middle of the sensor modules.
- Removed unused `vertex_x`, `vertex_y`, `vertex_z` steering parameters.

ENDRELEASENOTES

### Description
This is the fix for the #50 and small optimization mentioned here #54.

Everything is carefully wrapped into `if (_subDetName == "SET")` to not affect reconstruction of the VXD/SIT/FTD hits until further investigation if those are affected and any corrections are needed.

### Note
Current change would be impossible for the FTD because FTD strips are perpendicular to the `z` which would give infinity in division by `v[2]`. So even if correction is needed there, then the math would be different.
Not sure what happens inside VXD or SIT...

### Control plots

TODO

### CODE REVIEW

**Comment by dudarboh** [comment]
The current pull request improves the efficiency, but does not seem to solve the issue completely...

I attach intermediate control plot and will do further investigation...
c1_n2

**Comment by dudarboh** [comment]
The other missing space points seem to happen, because of the assumption of the straight track line from the IP, as @gaede has already suspected...

Here is the one example:
If one calculates intersection point, with respect to the straight line coming from the IP, one gets space point which exceeds the boundary of the sensor and is being reasonably rejected, as firing strips are too far away for such hit to exist if the actual track was coming from the IP.  (red square - rejected space point)

One can see that the track comes from very different angle from which it is possible to fire these strips simultaneously.
(large blue squares - SET SimHits).

image

From discussions with @gaede, ideally one would need to take into account helix direction to produce proper space point, however, it is not trivial how to do so, as `DDSpacePointBuilder` is executed prior to the track fit processors..

Also there are hits, that could be fixed by slightly increasing the tolerance... But one need to take care not to include too many ghost hits, which may be not very properly defined inside `DDSpacePointBuilder`

File: source/Digitisers/src/DDPlanarDigiProcessor.cc [hunk 1]
```diff
@@ -342,11 +342,26 @@ void DDPlanarDigiProcessor::processEvent( LCEvent * evt ) {
...
+        dd4hep::rec::Vector3D newPosTmp;
+        if (! _isStrip){
+            newPosTmp = (1./dd4hep::mm) * ( surf->localToGlobal( dd4hep::rec::Vector2D( (uL+uSmear)*dd4hep::mm, (vL+vSmear)*dd4hep::mm ) ) );
+        }
+        else if (_subDetName == "SET"){
+            newPosTmp = dd4hep::rec::Vector3D(xStripPos, yStripPos, zStripPos);
+        }
+        else{
+            newPosTmp = (1./dd4hep::mm) * ( surf->localToGlobal( dd4hep::rec::Vector2D( (uL+uSmear)*dd4hep::mm, 0. ) ) );
+        }
```
Comment (tmadlener) [inline source/Digitisers/src/DDPlanarDigiProcessor.cc:364]: For even better readability, you could reorganize this to: 
```cpp
dd4hep::rec::Vector3D newPosTmp;
if (_isStrip) {
  if (_subDetName == "SET") {
    // caluclate the smeared position and assign it for the SET
  } else {
   newPosTmp = (1./dd4hep::mm) * ( surf->localToGlobal( dd4hep::rec::Vector2D( (uL+uSmear)*dd4hep::mm, 0. ) ) );
} else {
  newPosTmp = newPosTmp = (1./dd4hep::mm) * ( surf->localToGlobal( dd4hep::rec::Vector2D( (uL+uSmear)*dd4hep::mm, (vL+vSmear)*dd4hep::mm ) ) );
}
```
(would also limit the scope of the `stripPos` variables

Comment (tmadlener) [inline source/Digitisers/src/DDPlanarDigiProcessor.cc:364]: For even better readability, you could reorganize this to: 
```cpp
dd4hep::rec::Vector3D newPosTmp;
if (_isStrip) {
  if (_subDetName == "SET") {
    // caluclate the smeared position and assign it for the SET
  } else {
   newPosTmp = (1./dd4hep::mm) * ( surf->localToGlobal( dd4hep::rec::Vector2D( (uL+uSmear)*dd4hep::mm, 0. ) ) );
} else {
  newPosTmp = newPosTmp = (1./dd4hep::mm) * ( surf->localToGlobal( dd4hep::rec::Vector2D( (uL+uSmear)*dd4hep::mm, (vL+vSmear)*dd4hep::mm ) ) );
}
```
(would also limit the scope of the `stripPos` variables

**Review by tmadlener — COMMENTED** [review]
As discussed already yesterday during the meeting: this fixes the issue of the SET strips not being aligned with module boundaries and can be merged. I only have a minor comment for code readability below, if you have time please fix it, but it is not extremely important.

The second issue you mention above with the assumption of tracks coming from the IP for making space points, needs to be addressed in another way.

Checking if things look (even) better without low pT tracks, is I think the only thing that is still missing before we can merge.

**Comment by dudarboh** [comment]
Cross check plot before/after fixing digitized strip positions for SimHits with momentum > 5 GeV only :
set_ineff_5gev

Still efficiency drop is present, but in the very close proximity to the border, which one probably would expect?...
"Wide" efficiency drop vanishes, which I think supports reasoning in the comments above

**Review by tmadlener — APPROVED** [review]

**Pull request merged**